api/tests/test_views.py

```diff
@@ -11,3 +12,35 @@ def test_login(self):
         )
 
         self.assertEqual(response.status_code, 200)
+
+
+class SignUpTest(TestSetUp):
+    def test_signup(self):
+        data = {
+            "username": "user2",
+            "email": "user2@gmail.com",
+            "password1": "user2abcqwe",
+            "password2": "user2abcqwe",
+            "first_name": "2",
+            "last_name": "user",
+        }
+        response = self.client.post(path=reverse("signup"), data=data, format="json")
+        res_data = response.data.get("user")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(res_data["username"], data["username"])
+        self.assertEqual(res_data["email"], data["email"])
+        self.assertEqual(res_data["first_name"], data["first_name"])
+        self.assertEqual(res_data["last_name"], data["last_name"])
+
+    def test_signup_with_invalid_data(self):
```

Comment:
thêm case test cho các quan điểm test khác em nhé
Invalid data: username blank, email blank, email invalid format, password2 not match, ...

-> ko chỉ so sánh mỗi status code, nên so sánh cả message nữa e